tests/framework/prototype.py

```diff
@@ -192,6 +196,25 @@ def assert_rule(self, custom_resource=None, cr_namespace='default'):
             " python -c \"import ray; ray.init(); print(ray.cluster_resources())\"")
         assert rtn == 0
 
+class CurlServiceRule(Rule):
+    """"Using curl to access the deployed application on Ray service"""
+    def assert_rule(self, custom_resource=None, cr_namespace='default'):
+        # Create a pod for running curl command, because the service is not exposed.
+        os.system("kubectl run curl --image=radial/busyboxplus:curl --command --\
+           /bin/sh -c \"while true; do sleep 10;done\"")
+        while True:
+            resp = client.CoreV1Api().read_namespaced_pod(name="curl",
+                                                    namespace='default')
+            if resp.status.phase != 'Pending':
+                break
+            time.sleep(1)
+        output = subprocess.check_output("kubectl exec curl -- \
```

Comment:
What's the comparison between `os.system` and `subprocess`?

Reply:
```os.system``` executes the command in a subshell, while ```subprocess``` can both execute the command in a shell or spawn a process directly. Besides, ```subprocess``` is able to retrieve the output, while the output in ```os.system``` is sent to the interpreter standard output stream. General speaking, ```subprocess``` provides more powerful facilities than ```os.system```, and it is mentioned in the [doc](https://docs.python.org/3/library/os.html#os.system) of ```os.sytem``` that using ```subprocess``` is preferable to using ```os.system```.

Reply:
Can we replace `os.system` with `subprocess`?